src/utils/auth_utils.js

```diff
@@ -22,17 +30,33 @@ const firebaseConfig = {
 const app = initializeApp(firebaseConfig);
 const auth = getAuth(app);
 
+/**
+ * Signs a user in with Google using Firebase
+ * @returns A boolean indicating whether or not the user is new
+ */
 const signInWithGoogle = async () => {
-  // TODO
+  try {
+    const provider = new GoogleAuthProvider();
+    const userCredential = await signInWithPopup(auth, provider);
+
+    return getAdditionalUserInfo(userCredential).isNewUser;
+  } catch (err) {
+    return err;
```

Comment:
change from returning to throwing the error. also this method will take two redirect paths depending on if user is new or not